test/equatable_test.dart

```diff
@@ -27,7 +27,7 @@ class SimpleEquatable<T> extends Equatable {
   List<Object> get props => [data];
 }
 
-class MultipartEquatable<T> extends Equatable {
+class MultipartEquatable<T extends Object> extends Equatable {
```

Comment:
I'm curious... Why is `extends Object` necessary here?

Reply:
It seems without it I'd need to make `props` return `List<Object?>`

![Screen Shot 2020-09-15 at 11 45 45 AM](https://user-images.githubusercontent.com/8855632/93240203-753fa500-f749-11ea-89aa-8edb2c12dedd.png)

```dart
class SimpleEquatable<T> extends Equatable {
  const SimpleEquatable(this.data);

  final T data;

  @override
  List<Object?> get props => [data];
}
```

Am I doing anything wrong and what would you suggest?

Reply:
I think that's correct. By `<T extends Object>`, you're saying "I want something non-nullable here". If Felix said just `<T>`, that would be equivalent to saying `<T extends dynamic>`, which is to say, "anything". Even nullable types.

I'm not 100% sure, though, so I'd love to hear @lrhn or @munificent's take on it.

Reply:
Your explanation makes sense, @felangel. I just hadn't thought through that case... I'd thought of the default case being non-nullability (`Object`), not `dynamic`/`Object?`.

Reply:
@felangel is right. The default bound on generic classes is `Object?`. That, by default, lets you create instances of the generic class using any type, even nullable ones. For example, the default bound of `Object?` on `List<T>` is what lets you write `new List<int?>`.

Using `extends Object` as the bound means that you can instantiate the generic class only with non-nullable types.

In most cases, the default bound is what you want. `extends Object` prohibits nullable types, but doesn't otherwise tell you anything particularly useful about the type parameter inside the body of the generic class. It's not like, say, `extends Widget` where then inside the class you can access all of the Widget methods on variables of the type parameter's type.